src/vcpkg/commands.extract.cpp

```diff
@@ -0,0 +1,166 @@
+            {
+                return {StripMode::Automatic, -1};
+            }
+
+            auto maybe_strip_value = Strings::strto<int>(maybe_value);
+            if (auto value = maybe_strip_value.get(); value && *value >= 0)
+            {
+                return {StripMode::Manual, *value};
+            }
+
+            // If the value is not an integer or is less than 0
+            msg::println_error(msgErrorInvalidExtractOption, msg::option = OPTION_STRIP, msg::value = maybe_value);
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+        // No --strip set, default to 0
+        return {StripMode::Manual, 0};
+    }
+
+    constexpr IsSlash is_slash;
+    size_t get_common_directories_count(std::vector<Path> paths)
+    {
+        if (paths.size() == 0)
+        {
+            return 0;
+        }
+
+        std::string known_common_prefix = paths[0].native();
+        for (size_t idx = 1; idx < paths.size(); ++idx)
+        {
+            auto&& candidate = paths[idx].native();
+            auto mismatch_point = std::mismatch(known_common_prefix.begin(),
+                                                known_common_prefix.end(),
+
+                                                candidate.begin(),
+                                                candidate.end())
+                                      .first;
+            known_common_prefix.erase(mismatch_point, known_common_prefix.end());
+        }
+
+        return std::count_if(known_common_prefix.begin(), known_common_prefix.end(), is_slash);
+    }
+
+    std::vector<std::pair<Path, Path>> get_archive_deploy_operations(const ExtractedArchive& archive,
+                                                                     StripSetting strip_setting)
+    {
+        std::vector<std::pair<Path, Path>> result;
+
+        const auto temp_dir = archive.temp_path;
+        const auto base_path = archive.base_path;
+        const auto proximate = archive.proximate_to_temp;
+
+        size_t strip_count = strip_setting.mode == StripMode::Automatic ? get_common_directories_count(proximate)
+                                                                        : static_cast<size_t>(strip_setting.count);
+
+        for (const auto& prox_path : proximate)
+        {
+            const auto& prox_str = prox_path.native();
+            auto first = prox_str.data();
+            auto last = first + prox_str.size();
+
+            // strip leading directories equivalent to the number specified
+            for (size_t i = 0; i < strip_count && first != last; ++i)
+            {
+                first = std::find_if(first, last, is_slash);
+                first = std::find_if_not(first, last, is_slash);
+            }
+
+            if (first == last)
+            {
+                continue;
+            }
+
+            result.emplace_back(temp_dir / Path{prox_path},
+                                Path{base_path} / std::string(first, static_cast<size_t>(last - first)));
```

Comment:
```suggestion
            result.emplace_back(temp_dir / prox_path,
                                base_path / std::string(first, static_cast<size_t>(last - first)));
```

`prox_path` and `base_path` are already `Path`. Maybe also consider replacing `auto` with the actual type to avoid confusion here.